### PR TITLE
Add a ports and protocols reference page

### LINKED ISSUE #29698: Add a reference doc for ports and protocols info
**What would you like to be added**
Create a dedicated ports and protocols page for Kubernetes Clusters. 

**Why is this needed**
- Currently, the closest page that provides this information is under KubeAdm install setup [page](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#check-required-ports). 
- This is hard to find for any cluster administrator trying to deploy Kubernetes
- Documented Ports for [kube-controller](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/) and [kube-scheduler](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-scheduler/) are incorrect and need an update
Slack thread: https://kubernetes.slack.com/archives/C1J0BPD2M/p1628568604069300

/sig security node docs architecture
/area security

### PR DESCRIPTION
**Pull request by PushkarJ**
- Refactored ports and protocols info under cluster-adminstration
- Updated the ports for kube-scheduler and kube-controller based on
  current state

Fixes #29698 

/sig security node architecture

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
Hm, extracting this information ouside of the kubeadm page makes sense to me.

So overall LGTM, thank you.

/sig cluster-lifecycle

**Comment by PushkarJ** [comment]
@sftim @neolit123 Ready for review

File: content/en/docs/reference/_index.md
Comment (sftim) [inline content/en/docs/reference/_index.md:67]:
```suggestion
* List of [ports and protocols](/docs/reference/ports-and-protocols/) that
```

File: content/en/docs/reference/ports-and-protocols.md
Comment (sftim) [inline content/en/docs/reference/ports-and-protocols.md:3]:
```suggestion
content_type: reference
```

File: content/en/docs/reference/ports-and-protocols.md
Comment (sftim) [inline content/en/docs/reference/ports-and-protocols.md:12]:
```suggestion
## Control plane
```
These ports need to be open even if you run the control plane directly (not within a kubelet).

File: content/en/docs/reference/ports-and-protocols.md
Comment (sftim) [inline content/en/docs/reference/ports-and-protocols.md:22]: I'd set the fragment identifier to `#node` because I want to pick something stable. People will want to hyperlink here.

```suggestion
## Worker node(s) {#node}
```

File: content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md
Comment (sftim) [inline content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:71]:
```suggestion
[required ports](/docs/reference/ports-and-protocols/)
```

File: content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md
Comment (sftim) [inline content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:72]: nit:
```suggestion
need to be open in order for Kubernetes components to communicate with each other.
```

File: content/en/docs/reference/ports-and-protocols.md
Comment (sftim) [inline content/en/docs/reference/ports-and-protocols.md:35]: - Maybe put that before the “Worker node(s)” heading?
- optionally: hyperlink to `/docs/setup/production-environment/tools/kubeadm/ha-topology/#external-etcd-topology`

Comment (sftim) [reply content/en/docs/reference/ports-and-protocols.md:35]: We could also put that comment in the “Check required ports” section of the kubeadm setup page. Your call I think.

**Review by sftim — CHANGES_REQUESTED** [review]
Thanks @PushkarJ 

Here are a few fixes along with optional suggestions. The main one you need to make is to fix some of the hyperlinks.

File: content/en/docs/reference/ports-and-protocols.md
Comment (PushkarJ) [reply content/en/docs/reference/ports-and-protocols.md]: Good call on placement. Moved under control plane section

**Review by sftim — COMMENTED** [review]
As this is mostly a migration, I'm happy to skip tech review.

/lgtm

**Comment by sftim** [comment]
[Preview](https://deploy-preview-29699--kubernetes-io-main-staging.netlify.app/docs/reference/ports-and-protocols/) of the new page.

File: content/en/docs/reference/ports-and-protocols.md
Comment (sftim) [inline content/en/docs/reference/ports-and-protocols.md:35]: - This feels more like advice specifically for people running kubeadm. I think if you deploy by hand, you can override any of these ports.
- (could be a follow up PR) we might want to explain that although the API server listens on TCP 6443 by default, typically you either:
  - change this to 443
  - put the API server behind a load-balancer

File: content/en/docs/reference/ports-and-protocols.md
Comment (sftim) [inline content/en/docs/reference/ports-and-protocols.md:18]: The control plane only listens on 10250 if it's running on a node. There are Kubernetes clusters that pass all the tests where the control plane hosts aren't listening on this port.

Comment (sftim) [reply content/en/docs/reference/ports-and-protocols.md:18]: We could add a caveat about this after “Although etcd ports are included in control plane section, you can also host your own etcd cluster externally or on custom ports.”

Comment (PushkarJ) [reply content/en/docs/reference/ports-and-protocols.md:18]: Does this come under general best practice, of "ports that are not used should not be open" ? Wondering if adding more caveats like this could confuse the reader.

File: content/en/docs/reference/ports-and-protocols.md
Comment (PushkarJ) [reply content/en/docs/reference/ports-and-protocols.md:35]: Updated the text. Would need an `/lgtm` again

File: content/en/docs/reference/ports-and-protocols.md
Comment (sftim) [reply content/en/docs/reference/ports-and-protocols.md:18]: :shrug: I don't think we have even weak guidelines on that. Happy for you to decide here @PushkarJ !

**Pull request merged**
